src/pk_core.c

```diff
@@ -606,13 +606,32 @@ DEF(coreInput,
   RET(VAR_OBJ(line));
 }
 
+DEF(coreExit,
+  "exit([value:int]) -> null\n"
+  "Sets the VM's fiber to NULL, causing the VM to return from execution "
+  "and thereby terminating the process entirely. The exit code of the "
+  "process is the optional argument [value] given to exit(). If no argument "
+  "is given, the exit code is 1 by default.") {
+
+  int argc = ARGC;
+  if (argc > 1) { // exit() or exit(val).
+    RET_ERR(newString(vm, "Invalid argument count."));
+  }
+
+  int64_t value;
+  if (!validateInteger(vm, ARG(1), &value, "Argument 1")) return;
```

Comment:
If the argc is 0, this call will cause UB, you need to check the `argc` before.
```c
int64_t value = 0;
if (argc == 1) {
  if (!validateInteger(vm, ARG(1), &value, "Argument 1")) return;
}
// TODO: this actually needs to be the VM fiber being set to null though.
exit(value);
```

Reply:
I've added those changes and more (see comment after push).